rails/DEPLOYMENTS.md

```diff
@@ -0,0 +1,53 @@
+
+Abtion Rails deployment environment guidelines:
+
+Draft v0.1
+Authors: 
+	Geoff Hubbard gh@abtion.com
+	Heather Moore hm@abtion.com
+
+Are we all-in on Heroku?
+
+CI:
+Github CI?
+Travis CI?
+Heroku CI
+
+Pipelines for promoting apps from staging to production.
```

Comment:
Are you actually promoting staging to production, or are you setting staging to autodeploy your development branch, and production to autodeploy your master branch?
Does promoting have other effects than just updating the code? (add-ons, env variables, etc?)

Reply:
:+1: @prrrnd 
I prefer having only `master`, which deploys to staging, then promoting staging to production (could be automated by using tags).
Mainly because it avoids accidents where commit exist on master, but not on develop.

> Does promoting have other effects than just updating the code? (add-ons, env variables, etc?)

Promoting, takes the code from staging and runs it with the env variables and add-ons that are on production.
What it doesn't do is recompile the code. So it will fail if you have some css/js compilation that differs based on environment variables.

Reply:
> What it doesn't do is recompile the code. So it will fail if you have some css/js compilation that differs based on environment variables.

Thanks @runephilosof, I think this is an important point that should be taken into consideration for JS-heavy projects.

Reply:
It is possible to build JS applications that can be deployed this way too.